ml-proto/host/lexer.mll

```diff
@@ -88,18 +88,19 @@ let escape = ['n''t''\\''\'''\"']
 let character =
   [^'"''\\''\x00'-'\x1f''\x7f'] | '\\'escape | '\\'hexdigit hexdigit
 
-let sign = ('+' | '-')?
-let num = sign digit+
-let hexnum = sign "0x" hexdigit+
-let int = num | hexnum
+let sign = ('+' | '-')
+let num = digit+
+let hexnum = "0x" hexdigit+
+let nat = num | hexnum
+let int = sign nat
```

Comment:
Would it make sense to make `int` be `sign? nat`, so that productions like `literal` don't have to match both `int` and `nat`?

Also, what is `nat` short for?